src/GeneratedMapper/Builders/ProjectToExtensionsBuilder.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using GeneratedMapper.Enums;
+using GeneratedMapper.Extensions;
+using GeneratedMapper.Information;
+using Microsoft.CodeAnalysis.Text;
+
+namespace GeneratedMapper.Builders
+{
+    internal sealed class ProjectToExtensionsBuilder
```

Comment:
This object is pretty similar to `MapToExtensionsBuilder`. I think most shared coded can be put in a base class and only have the mapto or projectto specifics be in the derived classes.

Reply:
I put a small base with not that much code share between them.  Only really the IndentWriter creation, which is so commonly used it should probably be its own Extension method so you don't repeat conditional operation in its constructor.
Maybe a wrapper that passes indent writer but can dispose of the String writer at the same time since the indent writer doesn't dispose of string writer when its Disposed.  That way there's only one writer instead of two at the start of writing files.

As for sharing other code I didn't want to attempt it since it would break up the flow of the spacing matching code if you hid it.  Sure you could refactor the code in the foreach loops into easier to read functions to make the indentWriter.WriteLine more readable, but how much does it really help?

Reply:
Ok. I'll have a look at after merging, see if I can perhaps improve on it.